libavcodec/x86/vvc_itx_1d.asm

```diff
@@ -0,0 +1,1142 @@
+;******************************************************************************
+;*
+;* SIMD-optimized inverse transform functions for VVC decoding
+;*
+;* Copyright (c) 2023 Frank Plowman <post@frankplowman.com>
+;*
+;* This file is part of FFmpeg.
+;*
+;* FFmpeg is free software; you can redistribute it and/or
+;* modify it under the terms of the GNU Lesser General Public
+;* License as published by the Free Software Foundation; either
+;* version 2.1 of the License, or (at your option) any later version.
+;*
+;* FFmpeg is distributed in the hope that it will be useful,
+;* but WITHOUT ANY WARRANTY; without even the implied warranty of
+;* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+;* Lesser General Public License for more details.
+;*
+;* You should have received a copy of the GNU Lesser General Public
+;* License along with FFmpeg; if not, write to the Free Software
+;* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+;*
+;******************************************************************************
+
+%include "libavutil/x86/x86util.asm"
+
+SECTION_RODATA
+
+const vvc_dct2_4_even_mat, dw 64, 64, 64, -64, 64, -64, 64, 64
+
+const vvc_dct2_4_odd_mat, dw 83, 36, 36, -83, -36, 83, -83, -36
+
+%define matvec_mul_4_permute(m11, m12, m13, m14, \
+                             m21, m22, m23, m24, \
+                             m31, m32, m33, m34, \
+                             m41, m42, m43, m44) \
+    m11, m12, m21, m22, m31, m32, m41, m42, \
+    m13, m14, m23, m24, m33, m34, m43, m44
+
+%define dct2_8_odd_mat_permute(m11, m12, m13, m14, \
+                               m21, m22, m23, m24, \
+                               m31, m32, m33, m34, \
+                               m41, m42, m43, m44) \
+    m41, m42, m43, m44, \
+    m31, m32, m33, m34, \
+    m21, m22, m23, m24, \
+    m11, m12, m13, m14
+    
+const vvc_dct2_8_odd_mat, dw matvec_mul_4_permute(dct2_8_odd_mat_permute( \
```

Comment:
vvc_itx_1d.asm:49: warning: single-line macro `matvec_mul_4_permute' exists, but not taking 1 parameter [-w+pp-macro-params-single]

Reply:
I think this is a bug in NASM - the inner macro is expanded. These macros could be removed and these permutations applied directly to the transform matrices at the cost of readability, or we could look at adding `-w-pp-macro-params-single`, if we really want to get rid of the warning.